src/java/org/apache/cassandra/gms/Gossiper.java

```diff
@@ -1986,20 +1986,15 @@ void buildSeedsList()
     /**
      * JMX interface for triggering an update of the seed node list.
      */
-    public List<String> reloadSeeds()
+    public synchronized List<String> reloadSeeds()
```

Comment:
why `synchronized`?  We don't do that when touching the `seeds` object so are you just trying to block concurrent nodetool commands?  I am cool with that, but to be 100% clear this does not fix the race condition bug with this method mutating `seeds`

Reply:
yes, blocking concurrent nodetool / jmx. What's the problem? How it doesnt fix it? Genuinely curious. Maybe I forgot something.

Reply:
I think it is because there are other accesses to `seeds` outside of this method, e.g when we remove seed when node is removed etc ... ah damn

Reply:
Also note that a TON of people have broken gossip by adding and removing synchronization.  I haven't seen it in the seed path, but elsewhere in gossiper, folks spent far, far too much time (pre-simulator) on this. If you touch synchronization in this area, you may want to see if the simulator can prove it doesn't cause problems.

Reply:
I may go without `synchronized` indeed. I dont think that in practice this would be problematic to not have synchronized.  If somebody reloads at the same time via different threads then this is not thread safe, but in practice, that would be true only in case somebody changes the content of cassandra.yaml for seeds list before the first invocation ends. The probability of that happening is practically zero.